// azapi/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "useRequest", {
  enumerable: true,
  get: function () {
    return _useRequest.default;
  }
});
Object.defineProperty(exports, "useQuery", {
  enumerable: true,
  get: function () {
    return _useQuery.default;
  }
});
Object.defineProperty(exports, "useBody", {
  enumerable: true,
  get: function () {
    return _useBody.default;
  }
});
Object.defineProperty(exports, "useHeaders", {
  enumerable: true,
  get: function () {
    return _useHeaders.default;
  }
});
Object.defineProperty(exports, "useHeader", {
  enumerable: true,
  get: function () {
    return _useHeader.default;
  }
});
Object.defineProperty(exports, "useParams", {
  enumerable: true,
  get: function () {
    return _useParams.default;
  }
});
Object.defineProperty(exports, "useEffect", {
  enumerable: true,
  get: function () {
    return _useEffect.default;
  }
});
Object.defineProperty(exports, "setStatus", {
  enumerable: true,
  get: function () {
    return _useStatus.default;
  }
});
Object.defineProperty(exports, "useState", {
  enumerable: true,
  get: function () {
    return _useState.default;
  }
});
Object.defineProperty(exports, "useCache", {
  enumerable: true,
  get: function () {
    return _useCache.default;
  }
});
Object.defineProperty(exports, "useContext", {
  enumerable: true,
  get: function () {
    return _useContext.default;
  }
});
Object.defineProperty(exports, "UNSAFE_createContext", {
  enumerable: true,
  get: function () {
    return _useContext.UNSAFE_createContext;
  }
});
Object.defineProperty(exports, "useAuthorization", {
  enumerable: true,
  get: function () {
    return _useAuthorization.default;
  }
});
Object.defineProperty(exports, "useContentType", {
  enumerable: true,
  get: function () {
    return _useContentType.default;
  }
});
Object.defineProperty(exports, "UNSAFE_setEnv", {
  enumerable: true,
  get: function () {
    return _env.UNSAFE_setEnv;
  }
});
Object.defineProperty(exports, "useEnv", {
  enumerable: true,
  get: function () {
    return _env.useEnv;
  }
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function () {
    return _api.default;
  }
});
Object.defineProperty(exports, "MemoryCacheStrategy", {
  enumerable: true,
  get: function () {
    return _MemoryCacheStrategy.default;
  }
});

var _useRequest = _interopRequireDefault(require("./hooks/useRequest"));

var _useQuery = _interopRequireDefault(require("./hooks/useQuery"));

var _useBody = _interopRequireDefault(require("./hooks/useBody"));

var _useHeaders = _interopRequireDefault(require("./hooks/useHeaders"));

var _useHeader = _interopRequireDefault(require("./hooks/useHeader"));

var _useParams = _interopRequireDefault(require("./hooks/useParams"));

var _useEffect = _interopRequireDefault(require("./hooks/useEffect"));

var _useStatus = _interopRequireDefault(require("./hooks/useStatus"));

var _useState = _interopRequireDefault(require("./hooks/useState"));

var _useCache = _interopRequireDefault(require("./hooks/useCache"));

var _useContext = _interopRequireWildcard(require("./hooks/useContext"));

var _useAuthorization = _interopRequireDefault(require("./hooks/useAuthorization"));

var _useContentType = _interopRequireDefault(require("./hooks/useContentType"));

var _env = require("./env");

var _api = _interopRequireDefault(require("./api"));

var _MemoryCacheStrategy = _interopRequireDefault(require("./cachingStrategies/MemoryCacheStrategy"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
