// azapi/dist/hooks/useContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = useContext;
exports.applyInterface = applyInterface;
exports.UNSAFE_createContext = UNSAFE_createContext;

var _iscope = _interopRequireDefault(require("iscope"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const useContextScope = (0, _iscope.default)(() => null);

function useContext(context, func) {
  if (!arguments.length) {
    return useContextScope();
  }

  applyInterface(context);
  return useContextScope(context, func);
}

function applyInterface(context) {
  if (context.__appliedInterface) {
    return context;
  }

  context.__appliedInterface = true;
  const on = {};
  Object.assign(context, {
    on(name, handler) {
      let handlers = on[name];

      if (!handlers) {
        on[name] = handlers = new Set();
      }

      handlers.add(handler);
      return () => handlers.delete(handler);
    },

    emit(name, params = context) {
      const handlers = on[name];

      if (!handlers) {
        return;
      }

      return Array.from(handlers).reduce((prev, handler) => {
        if (prev && typeof prev.then === 'function') {
          return prev.finally(() => {
            return handler(params);
          });
        } else {
          return handler(params);
        }
      }, null);
    },

    dispose() {
      context.emit('disposing');
    }

  });
  return context;
}

function UNSAFE_createContext(req, res, props) {
  return applyInterface({
    req,
    res,
    ...props
  });
}
